### PR TITLE
fix: shouldDoInterceptionBasedOnUrl returns true for any valid subdomain

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change

The `shouldDoInterceptionBasedOnUrl` function now returns true if `sessionTokenBackendDomain` is a valid subdomain of the URL's domain. This aligns with the behavior of browsers when sending cookies to subdomains.

## Related issues
- https://github.com/supertokens/supertokens-node/issues/826
- https://github.com/supertokens/supertokens-website/issues/217

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [x] Changelog has been updated
- [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
   - Along with the associated array in `lib/ts/version.ts`
- [x] Changes to the version if needed
   - In `package.json`
   - In `package-lock.json`
   - In `lib/ts/version.ts`
- [x] Had run `npm run build-pretty`
- [x] Had installed and ran the pre-commit hook
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,32 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  
+  ```javascript
+    shouldDoInterceptionBasedOnUrl("https://sub.api.example.com", "", "api.example.com") // true
+    shouldDoInterceptionBasedOnUrl("https://sub.api.example.com", "", ".api.example.com") // true
+    shouldDoInterceptionBasedOnUrl("https://sub.api.example.com", "", "example.com") // true
+    shouldDoInterceptionBasedOnUrl("https://sub.api.example.com", "", ".example.com") // true
+  ```
+
+The behavior in previous versions differed, as `shouldDoInterceptionBasedOnUrl` returned `false` unless `sessionTokenBackendDomain` had an explicit leading dot. This breaking change now assumes any subdomain is valid, making the interception consistent with browser cookie policies.
+
```
Comment (rishabhpoddar) [inline CHANGELOG.md:35]: the behaviour of sessionTokenFrontendDomain has also changed, correct? At least cause passing example.com and .example.com has the same effect now

File: lib/ts/recipeImplementation.ts [hunk 2]
```diff
@@ -303,11 +303,7 @@ export default function RecipeImplementation(recipeImplInput: {
                         domain = urlObj.port === "" ? domain : domain + ":" + urlObj.port;
                     }
                 }
-                if (sessionTokenBackendDomain.startsWith(".")) {
```
Comment (rishabhpoddar) [inline lib/ts/recipeImplementation.ts:306]: also fix the normaliseSessionScopeOrThrowError functino to remove the redundant check, similar to how it's done in the node sdk

File: test/config.test.js [hunk 3]
```diff
@@ -112,15 +120,12 @@ describe("Config tests", function () {
...
         assert(!shouldDoInterceptionBasedOnUrl("https://sub.api.example.com:3000", "", "localhost"));
         assert(!shouldDoInterceptionBasedOnUrl("http://127.0.0.1:3000", "", "https://127.0.0.1:3010"));
         assert(!shouldDoInterceptionBasedOnUrl(window.location.hostname, "", "localhost"));
         assert(!shouldDoInterceptionBasedOnUrl(window.location.hostname, "", ".localhost"));
         assert(!shouldDoInterceptionBasedOnUrl(window.location.hostname, "", "localhost:2000"));
-        assert(!shouldDoInterceptionBasedOnUrl("https://sub.api.example.co.uk:3000", "", "api.example.co.uk"));
-        assert(!shouldDoInterceptionBasedOnUrl("https://sub.api.example.co.uk", "", "api.example.co.uk"));
 
         // errors in input
         try {
```
Comment (rishabhpoddar) [inline test/config.test.js:131]: not related to this file, but need to add e2e tests for the whole duplicate cookie domain thing

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (anku255) [reply CHANGELOG.md:35]: No, there are no changes in `sessionTokenFrontendDomain`. The `normaliseSessionScopeOrThrowError` scope function had no functional changes in this PR.  We only removed a redundant leading dot check but that didn't affect anything.

File: lib/ts/recipeImplementation.ts (hunk 2 above)
Comment (anku255) [reply lib/ts/recipeImplementation.ts:306]: Done

File: test/config.test.js (hunk 3 above)
Comment (anku255) [reply test/config.test.js:131]: We decided to postpone it.

**Review by rishabhpoddar — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]
